File: app/src/main/java/com/nvvi9/giphyme/ui/viewmodels/FavoriteViewModel.java
package com.nvvi9.giphyme.ui.viewmodels;

import androidx.lifecycle.ViewModel;
import androidx.paging.PagingData;

import com.nvvi9.giphyme.domain.FavoriteGifsUseCase;
import com.nvvi9.giphyme.vo.FavoriteGifItem;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;
import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Flowable;

@HiltViewModel
public class FavoriteViewModel extends ViewModel {

    private final FavoriteGifsUseCase favoriteGifsUseCase;

    @Inject
    public FavoriteViewModel(FavoriteGifsUseCase favoriteGifsUseCase) {
        this.favoriteGifsUseCase = favoriteGifsUseCase;
    }

    public Flowable<PagingData<FavoriteGifItem>> getFavoriteGifItems() {
        return favoriteGifsUseCase.getFavorites();
    }

    public Completable deleteFromFavorites(String gifId) {
        return favoriteGifsUseCase.deleteFromFavorites(gifId);
    }
}

File: app/src/main/java/com/nvvi9/giphyme/ui/viewmodels/FavoriteViewModel.kt
package com.nvvi9.giphyme.ui.viewmodels

import androidx.lifecycle.ViewModel
import androidx.paging.PagingData
import com.nvvi9.giphyme.domain.FavoriteGifsUseCase
import com.nvvi9.giphyme.vo.FavoriteGifItem
import dagger.hilt.android.lifecycle.HiltViewModel
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Flowable
import javax.inject.Inject

@HiltViewModel
class FavoriteViewModel @Inject constructor(private val favoriteGifsUseCase: FavoriteGifsUseCase) :
    ViewModel() {

    val favoriteGifItems: Flowable<PagingData<FavoriteGifItem>>
        get() = favoriteGifsUseCase.getFavorites()

    fun deleteFromFavorites(gifId: String): Completable =
        favoriteGifsUseCase.deleteFromFavorites(gifId)
}